editor/plugins/script_editor_plugin.cpp

```diff
@@ -3424,6 +3426,11 @@ void ScriptEditor::_history_back() {
 	}
 }
 
+void ScriptEditor::_pin_pressed() {
+	bool current_value = EditorSettings::get_singleton()->get("text_editor/behavior/navigation/stay_in_script_editor_on_node_selected");
+	EditorSettings::get_singleton()->set("text_editor/behavior/navigation/stay_in_script_editor_on_node_selected", !current_value);
```

Comment:
You should save the settings, otherwise this won't be remembered.

Reply:
Thanks. I assumed it was already permanent.